packages/front-end/components/Settings/EditDataSource/DataSourceExperimentProperties/DataSourceEditExperimentEventPropertiesModal.tsx

```diff
@@ -0,0 +1,94 @@
+import { FC } from "react";
+import { useForm } from "react-hook-form";
+import {
+  DataSourceEvents,
+  DataSourceInterfaceWithParams,
+} from "back-end/types/datasource";
+import Field from "../../../Forms/Field";
+import Modal from "../../../Modal";
+
+type DataSourceEditExperimentEventPropertiesProps = {
+  dataSource: DataSourceInterfaceWithParams;
+  onSave: (experimentProps: DataSourceEvents) => void;
+  onCancel: () => void;
+};
+
+export const DataSourceEditExperimentEventPropertiesModal: FC<DataSourceEditExperimentEventPropertiesProps> = ({
```

Comment:
This component is for editing the type identified above as `DataSourceEvents`. If there's a better name for it, let me know.